### DISCUSSION TITLE
Using several HM WR in parallel/Huawei+HMT,HMS support/HMS XXX-W WiFi version support

### DISCUSSION THREAD

**Discussion by gitisgreat2023** [hoylabs/OpenDTU-OnBattery#599] [Q&A]
Some questions, they are related to each other.
1) Would it be feasible to use, say two HM WR in parallel? Say a HM 1500 and a HM 800, enabling 2300W in total. One can think of many implementations in software, but say the simplest solution of proportional power limiting is used (1150 W required, both are limited to 50%). How's the performance currently, how fast can be two HMs power limit be changed?
2) "Please note: Using the Huawei AC charger in combination with the CMT2300A radio board is not supported at the moment." --> so this means, with the Huawei AC charger at max the HM 1500 can be used, right?
3) "In contrast to the original openDTU, with release 2023.05.23.post1 openDTU-onBattery supports only 5 inverters." I don't find a link to these 5 inverters
4) There is a [first implementation](https://github.com/DennisOSRM/hms-mqtt-publisher/tree/main) of the [HMS WiFi version](https://www.hoymiles.com/product/hms-600w-700w-800w-900w-1000w-2t-wi-fi-integrated-eu/) (goese up to 1000W currently). Is it plausible that this is supported as well? Then using the WiFi the new WiFi enabled HM would be supported

**Comment by spcqike** [comment]
2:
the HM 800 also uses the NRF24L01+, so that shoudn't be a problem. the HMS instead needs the CMT2300A.

as for 3.:
all inverters, that are support by original openDTU, are supported by openDTU-onBattery. but you only can connect 5 instead of 10 inverters, as openDTU-onBattery needs ressources for openDTU-onBattery exclusive features like powermeter, battery, shunt, ac charger, ......

**Reply by gitisgreat2023** [reply]
on 2): yes I understand. The max wattage, when using the Huawei, is 1500 W right? Because the more powerfull models are using the sub 1G communication module, which cannot be used simultaneously with the Huawei. Do I understand that correctly?

On 3: is that 5 (or 10) _**simultaneously**_ ?

**Reply by spcqike** [reply]
yes, simultaneously. but only as DTU (to get inverter live values and stuff). 
dynamic power limiting only works for 1 inverter so far.

in your case, i would probably let the HMS 800 run freely and open and use the HM 1500 with openDTU and powerlimiting. i don't know how this works with huawei AC charger, as i never used one. but for your home consumption, one "open" HMS800 and in addition one limited HM1500 should work. 

i would like to know what happens, if your DTU sees like -300W (because the HMS800 produces to much, HM1500 is off). would it start charging with 300W to get 0 export?

**Reply by gitisgreat2023** [reply]
From the logic, yes, I the HM-800 runs full at 800, the system sees -300 the AC charger will charge 300. At the battery level thus 800 outflow sees 300 inflow, in principle for the 300 W "only" the losses in the HM and Huawei apply, say each 5%, so 30W.

**Comment by Manos1966** [comment]
> "Please note: Using the Huawei AC charger in combination with the CMT2300A radio board is not supported at the moment." --> so this means, with the Huawei AC charger at max the HM 1500 can be used, right?

The CMT2300A board is not used to communicate with the HM Series invenrters (they require a NRF24L01+ board).
Therefore, you will have no problem using the Huawei AC charger and the HM1500.
A full list of the Hoymiles models and their respective communication requirements you can find under:
https://github.com/tbnobody/OpenDTU#currently-supported-inverters


PS. Dammit @spcqike 🤣

**Comment by Manos1966** [comment]
> There is a [first implementation](https://github.com/DennisOSRM/hms-mqtt-publisher/tree/main) of the [HMS WiFi version](https://www.hoymiles.com/product/hms-600w-700w-800w-900w-1000w-2t-wi-fi-integrated-eu/) (goes up to 1000W currently). 

Thanks for this link.

> One can only fetch updates approximately twice per minute. The inverter firmware seems to implement a mandatory **wait period of a little more than 30 seconds.** If one makes a request within 30 seconds of the previous one, then the inverter will reply with the previous reading and restart the countdown. 

The 30second wait period gives me hopes for the future. The first comments I had read about the new WiFi inverters were talking about wait times of several minutes...
